base/applications/utilman/process.c

```diff
@@ -0,0 +1,189 @@
+            {
+                /* The process we are searching for is running */
+                bIsRunning = TRUE;
+                break;
+            }
+        }
+        while (Process32NextW(hSnapshot, &Process));
+    }
+
+    /* Free the handle and return */
+    CloseHandle(hSnapshot);
+    return bIsRunning;
+}
+
+/**
+ * @LaunchProcess
+ *
+ * Executes a process.
+ *
+ * @param[in]   lpProcessName
+ *     The name of the executable process.
+ *
+ * @return
+ *     Returns TRUE if the process has been launched successfully,
+ *     FALSE otherwise.
+ *
+ */
+BOOL LaunchProcess(LPCWSTR lpProcessName)
+{
+    STARTUPINFOW si;
+    PROCESS_INFORMATION pi;
+    HANDLE hUserToken, hProcessToken;
+    BOOL bSuccess;
+    WCHAR ExpandedCmdLine[MAX_PATH];
+
+    /* Expand the process path string */
+    ExpandEnvironmentStringsW(lpProcessName, ExpandedCmdLine, ARRAYSIZE(ExpandedCmdLine));
+
+    ZeroMemory(&pi, sizeof(pi));
+    ZeroMemory(&si, sizeof(si));
+    si.cb = sizeof(si);
+    si.dwFlags = STARTF_USESHOWWINDOW;
+    si.wShowWindow = SW_SHOWNORMAL;
+
+    /* Get the token of the parent (current) process of the application */
+    bSuccess = OpenProcessToken(GetCurrentProcess(), TOKEN_QUERY | TOKEN_DUPLICATE, &hUserToken);
+    if (!bSuccess)
+    {
+        DPRINT("OpenProcessToken() failed with error -> %lu\n", GetLastError());
+        return FALSE;
+    }
+
+    /* Duplicate a new token so that we can use it to create our process */
+    bSuccess = DuplicateTokenEx(hUserToken, TOKEN_ALL_ACCESS, NULL, SecurityIdentification, TokenPrimary, &hProcessToken);
+    if (!bSuccess)
+    {
+        DPRINT("DuplicateTokenEx() failed with error -> %lu\n", GetLastError());
+        CloseHandle(hUserToken);
+        return FALSE;
+    }
+
+    /* Finally create the process */
+    bSuccess = CreateProcessAsUserW(hProcessToken,
+                                    NULL,
+                                    ExpandedCmdLine,
+                                    NULL,
+                                    NULL,
+                                    FALSE,
+                                    0, // DETACHED_PROCESS, NORMAL_PRIORITY_CLASS
+                                    NULL,
+                                    NULL,
+                                    &si,
+                                    &pi);
+
+    if (!bSuccess)
+    {
+        DPRINT("CreateProcessAsUserW() failed with error -> %lu\n", GetLastError());
+        CloseHandle(hUserToken);
+        CloseHandle(hProcessToken);
+        return FALSE;
+    }
+
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+    CloseHandle(hUserToken);
+    CloseHandle(hProcessToken);
+    return TRUE;
+}
+
+/**
+ * @CloseProcess
+ *
+ * Closes a process.
+ *
+ * @param[in]   lpProcessName
+ *     The name of the executable process.
+ *
+ * @return
+ *     Returns TRUE if the process has been terminated successfully,
+ *     FALSE otherwise.
+ *
+ */
+BOOL CloseProcess(IN LPCWSTR lpProcessName)
```

Comment:
Something to refactor on the next PR for this tool:
You can largely combine `IsProcessRunning` and `CloseProcess`,  by creating one helper function that simply finds the process ID of the process you are interested, and returns that (or 0 if it's not found).
Then both functions do not need to walk the process list exactly the same way.